apps/site/components/Button/ButtonLoading.tsx

```diff
@@ -0,0 +1,15 @@
+import React, { useState } from 'react'
+import { Button } from '@faststore/ui'
+
+const ButtonLoading = () => {
+  const [loading, setLoading] = useState(false)
+
+  return (
+    <Button variant="primary" onClick={() => { setLoading(true) }} loadingLabel="Loading" loading={loading}>
+      Click me to trigger loading
+    </Button>
+  )
+
+}
```

Comment:
What do you think?


https://user-images.githubusercontent.com/10627086/212949574-4fd73379-2473-4352-bd64-a413a45524af.mov




```suggestion
import React, { useEffect, useState } from 'react'
import { Button } from '@faststore/ui'

const ButtonLoading = () => {
  const [loading, setLoading] = useState(false)

  useEffect(() => {
    let timer = null

    if (loading) {
      timer = window.setInterval(() => setLoading(false), 2000)
    }

    return () => {
      window.clearInterval(timer)
    }
  }, [loading])

  return (
    <Button
      variant="primary"
      onClick={() => setLoading(true)}
      loadingLabel="Loading"
      loading={loading}
    >
      Click me to trigger loading
    </Button>
  )
}
```